sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -391,12 +396,6 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
         )
         df.sparkSession.sessionState.executePlan(
           CreateTable(tableDesc, mode, Some(df.logicalPlan))).toRdd
-        if (tableDesc.partitionColumnNames.nonEmpty &&
```

Comment:
I think this is unnecessary since the create / insert commands already recover partitions themselves. We'll see if any tests fail.